### DISCUSSION TITLE
Callback after apply a suggestion

### DISCUSSION THREAD

**Discussion by fcsonline** [eslint/eslint#17094] [Q&A] [answered]
Hi,

Our team is working on creating a custom Eslint rule, but we have encountered an issue we cannot resolve. We coded a suggestion object that includes a fix method, but despite thoroughly reviewing the `fixer` and `fixing` documentation, we have been unable to determine how to execute a callback when the suggestion is applied.

Do you know a way to do it?

**Comment by mdjermanovic** [comment]
Hi @fcsonline!

Suggestions produce  [`EditInfo`](https://eslint.org/docs/latest/integrate/nodejs-api#-editinfo-type) objects containing info about the fixes, and then IDEs control when and how the fixes will be applied. There's no way to define a callback that would be executed when the suggestion is applied. For more details on why this feature isn't feasible, please see [RFC #39](https://github.com/eslint/rfcs/pull/93).

**Reply by fcsonline** [reply]
Thanks for sharing this information. It will help us to understand the reasons. I was trying to play with a eslint fork to add this callback but it looks like I was in the wrong path.

Do you know if more people are asking for this?

Just to add more context here, in our case, we were trying to code a fix function that instead of mutating the AST node text, it was modifying an external file. With this, we were "solving" the offense. But it seems impossible to do with the current implementation of the `fixer`.

**Reply by fcsonline** [reply]
I was trying something like this: https://github.com/eslint/eslint/compare/main...fcsonline:eslint:feature/fix-callback

**Reply by mdjermanovic** [reply, marked as answer]
> Do you know if more people are asking for this?
> 
> Just to add more context here, in our case, we were trying to code a fix function that instead of mutating the AST node text, it was modifying an external file. With this, we were "solving" the offense. But it seems impossible to do with the current implementation of the `fixer`.

I think this is a similar use case as in https://github.com/eslint/eslint/issues/16143.

> I was trying something like this: [main...fcsonline:eslint:feature/fix-callback](https://github.com/eslint/eslint/compare/main...fcsonline:eslint:feature/fix-callback)

ESLint produces suggestion objects (`{ desc: string; fix: EditInfo }`), but integrations (IDEs) don't use ESLint when applying the suggested fixes. When you choose to apply a suggestion in your IDE, IDE itself changes the text based on the previously provided [`EditInfo`](https://eslint.org/docs/latest/integrate/nodejs-api#-editinfo-type), so ESLint doesn't know if the suggestion was applied. According to https://github.com/eslint/rfcs/pull/93#issuecomment-1266454318, notifying ESLint that a suggestion was applied doesn't seem feasible for the VS Code ESLint integration.

**Reply by fcsonline** [reply]
Ok, got it! Thanks for all this comments! 💯
